operations/mimir/read-write-deployment/autoscaling.libsonnet

```diff
@@ -1 +1,37 @@
-{}
+{
+  _config+:: {
+    autoscaling_mimir_read_enabled: false,
+    autoscaling_mimir_read_min_replicas: error 'you must set autoscaling_mimir_read_min_replicas in the _config',
+    autoscaling_mimir_read_max_replicas: error 'you must set autoscaling_mimir_read_max_replicas in the _config',
+  },
+
+  //
+  // Helper methods
+  //
+
+  local removeReplicasFromSpec = {
+    spec+: {
+      // Remove the "replicas" field so that Flux doesn't reconcile it.
+      replicas+:: null,
+    },
+  },
+
+  read_scaled_object: if !$._config.autoscaling_mimir_read_enabled then null else
```

Comment:
[non blocking] The resulting metric name is `cortex_mimir_read_hpa_default`: looks a bit weird having `cortex_mimir`. As a follow up work we may consider getting rid of that `cortex_` prefix and only keep `mimir_`  prefix.